### PR TITLE
[VAL-791] Do not encode CDATA tags.

### PR DESCRIPTION
**Pull request by nathansf**
@kevintse328 @ktamas77 
Skip opening and closing CDATA tags when encoding. Wrapping xml values in CDATA tags for security reasons. Added test.

### CODE REVIEW

File: suds/sax/enc.py [hunk 1]
```diff
@@ -61,8 +61,15 @@ def encode(self, s):
         @rtype: str
         """
         if isinstance(s, basestring) and self.needsEncoding(s):
+            # Skip encoding of CDATA
+            cdata_found = False
+            if len(s) > 12 and '<![CDATA[' == s[:9] and ']]>' == s[-3:]:
```
Comment (kevintse328) [inline suds/sax/enc.py]: Please concatenate strings following our [Python style guide](https://prezidoc.atlassian.net/wiki/display/EN/Python+Style+Guide) by using `format`.

File: suds/sax/enc.py [hunk 2]
```diff
@@ -61,8 +61,15 @@ def encode(self, s):
...
+                s = s[9:-3]
+                cdata_found = True
             for x in self.encodings:
                 s = re.sub(x[0], x[1], s)
+            if cdata_found:
+                s = '<![CDATA[' + s + ']]>'
```
Comment (kevintse328) [inline suds/sax/enc.py]: See above.

**Comment by kevintse328** [comment]
:+1:

**Comment by ktamas77** [comment]
:+1:

**Pull request merged**